crates/wasm/tests/test_build.rs

```diff
@@ -0,0 +1,469 @@
+extern crate penumbra_wasm;
+
+#[cfg(test)]
+mod tests {
+    use anyhow::Result;
+    use serde::{Deserialize, Serialize};
+    use serde_json;
+    use wasm_bindgen::JsValue;
+    use wasm_bindgen_test::*;
+    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
+    use indexed_db_futures::prelude::{
+        IdbDatabase, IdbObjectStore, IdbQuerySource, IdbTransaction, IdbTransactionMode,
+    };
+
+    use penumbra_proto::{
+        core::{
+            asset::v1alpha1::Value,
+            component::chain::v1alpha1::{ChainParameters, FmdParameters},
+            keys::v1alpha1::{Address, AddressIndex},
+            transaction::v1alpha1::{MemoPlaintext, TransactionPlan as tp},
+        },
+        view::v1alpha1::SpendableNoteRecord,
+    };
+    use penumbra_tct::{structure::Hash, Forgotten};
+    use penumbra_transaction::{
+        plan::{ActionPlan, TransactionPlan},
+        Action,
+    };
+    use penumbra_wasm::{
+        error::WasmError,
+        storage::IndexedDBStorage,
+        tx::{authorize, build, build_parallel, witness},
+        wasm_planner::WasmPlanner,
+    };
+
+    #[wasm_bindgen_test]
+    async fn mock_build_serial_and_parallel() {
+        // Limit the use of Penumbra Rust libraries since we're mocking JS calls
+        // that are based on constructing objects according to protobuf definitions.
```

Comment:
Why can't we import the structs instead of re-defining them here?

Reply:
we'd need to mark the fields on the `IndexedDbConstants` and `Tables` structs as **pub**, since the test suite can't access the private fields.

Reply:
I think that would be a worthwhile tradeoff instead of re-writing the structs. What do you think?

Reply:
@hdevalence is there any issues with making these fields pub?